src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1122,39 +1125,133 @@ class spell_pri_power_word_shield : public SpellScriptLoader
+        // Target: player that currently has Prayer of Mending aura on him
+        Unit* target = GetTarget();
+        if (Unit* caster = GetCaster())
         {
-            PrepareSpellScript(spell_pri_prayer_of_mending_heal_SpellScript);
+            // Cast the spell to heal the owner
+            caster->CastSpell(target, SPELL_PRIEST_PRAYER_OF_MENDING_HEAL, true);
 
-            void HandleHeal(SpellEffIndex /*effIndex*/)
-            {
-                if (Unit* caster = GetOriginalCaster())
-                {
-                    if (AuraEffect* aurEff = caster->GetAuraEffect(SPELL_PRIEST_T9_HEALING_2P, EFFECT_0))
-                    {
-                        int32 heal = GetHitHeal();
-                        AddPct(heal, aurEff->GetAmount());
-                        SetHitHeal(heal);
-                    }
-                }
-            }
+            // Only cast jump if stack is higher than 0
+            int32 stackAmount = GetStackAmount();
+            if (stackAmount > 1)
+                target->CastCustomSpell(SPELL_PRIEST_PRAYER_OF_MENDING_JUMP, SPELLVALUE_BASE_POINT0, stackAmount - 1, target, true, nullptr, aurEff, caster->GetGUID());
 
-            void Register() override
+            Remove();
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectProc += AuraEffectProcFn(spell_pri_prayer_of_mending_aura::HandleHeal, EFFECT_0, SPELL_AURA_DUMMY);
+    }
+};
+
+// 155793 - prayer of mending (Jump) - SPELL_PRIEST_PRAYER_OF_MENDING_JUMP
+class spell_pri_prayer_of_mending_jump : public spell_pri_prayer_of_mending_SpellScriptBase
+{
+    PrepareSpellScript(spell_pri_prayer_of_mending_jump);
+
+    void OnTargetSelect(std::list<WorldObject*>& targets)
+    {
+        // Find the best target - prefer players over pets
+        Unit* owner = GetCaster();
+        bool foundPlayer = false;
+        for (WorldObject* worldObject : targets)
+        {
+            if (worldObject->IsPlayer())
             {
-                OnEffectHitTarget += SpellEffectFn(spell_pri_prayer_of_mending_heal_SpellScript::HandleHeal, EFFECT_0, SPELL_EFFECT_HEAL);
+                foundPlayer = true;
+                break;
             }
-        };
+        }
 
-        SpellScript* GetSpellScript() const override
+        if (foundPlayer)
+            targets.remove_if(Trinity::ObjectTypeIdCheck(TYPEID_PLAYER, false));
+
+        // choose one random target from targets
+        if (targets.size() > 1)
         {
-            return new spell_pri_prayer_of_mending_heal_SpellScript();
+            WorldObject* selected = Trinity::Containers::SelectRandomContainerElement(targets);
+            targets.clear();
+            targets.push_back(selected);
         }
+    }
+
+    void HandleJump(SpellEffIndex /*effIndex*/)
+    {
+        Unit* origCaster = GetOriginalCaster(); // the one that started the prayer of mending chain
+        Unit* target = GetHitUnit(); // the target we decided should jump the aura to
```

Comment:
This comment needs cleanup, poor sentence order.
```suggestion
        Unit* target = GetHitUnit(); // the target we decided the aura should jump to
```